client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/NearCache.java

```diff
@@ -10,9 +10,12 @@
  * @since 7.1
  */
 public interface NearCache<K, V> extends Iterable<Map.Entry<K, MetadataValue<V>>> {
-   void put(K key, MetadataValue<V> value);
-   void putIfAbsent(K key, MetadataValue<V> value);
+   boolean putIfAbsent(K key, MetadataValue<V> value);
```

Comment:
This is a public API since users can inject it via `org.infinispan.hotrod.configuration.NearCacheConfigurationBuilder#nearCacheFactory`. I know WF uses a custom implementation.

Reply:
I had wondered about that :(
But I don't think there is a way we can fix this otherwise, the side benefit is this is for a major release. But I can add a comment to the upgrading guide.